velox/functions/sparksql/window/tests/NthValueTest.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/lib/window/tests/WindowTestBase.h"
+#include "velox/functions/sparksql/window/WindowFunctionsRegistration.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::window::test {
+
+namespace {
+
+class SparkNthValueTest : public WindowTestBase {
+ protected:
+  void SetUp() override {
+    WindowTestBase::SetUp();
+    WindowTestBase::options_.parseIntegerAsBigint = false;
+    velox::functions::window::sparksql::registerWindowFunctions("");
+  }
+};
+
+TEST_F(SparkNthValueTest, basic) {
+  vector_size_t size = 20;
+
+  auto vectors = makeRowVector({
+      makeFlatVector<int32_t>(size, [](auto row) { return row; }),
+      makeFlatVector<int32_t>(size, [](auto row) { return row % 50; }),
+      makeFlatVector<int32_t>(size, [](auto row) { return row % 5 + 1; }),
+  });
+
+  std::string overClause = "partition by c0 order by c1";
+  std::string frameClausee = "rows between 1 preceding and current row";
```

Comment:
Is this the default spark frame ? Presto uses "range between unbounded preceding and current row" ?

Reply:
Thanks for pointing this, spark also uses "range between unbounded preceding and current row", updated the test cases.